src/Filters/filt.jl

```diff
@@ -105,7 +105,7 @@ function filt!(out::AbstractArray, f::Biquad{:z}, x::AbstractArray,
     (size(si, 1) != 2 || (N > 1 && Base.trailingsize(si, 2) != ncols)) &&
         throw(ArgumentError("si must have two rows and 1 or nsignals columns"))
 
-    for col = 1:ncols
+    for col in CartesianIndices(axes(x)[2:end])
         _filt!(out, si[1, N > 1 ? col : 1], si[2, N > 1 ? col : 1], f, x, col)
```

Comment:
No, if `N==1`, we only have a single initial state which we have to use for every `col`. If anything this could be
```suggestion
        _filt!(out, si[1, N > 1 ? col : CartesianIndex()], si[2, N > 1 ? col : CartesianIndex()], f, x, col)
```
but that doesn't seem any better to me.

That said, I'm a bit doubtful about the utility of this one-initial-state-for-all-columns behavior, but would subsume removing that in #599 if we decide we want to do that.